common/djangoapps/student/helpers.py

```diff
@@ -270,6 +262,31 @@ def get_next_url_for_login_page(request):
     return redirect_to
 
 
+def get_redirect_to(request):
+    """
+    Determine the redirect url and return if safe
+    :argument
+        request: request object
+
+    :returns: redirect url if safe else None
+    """
+    static_file_format = '\w+\.(jpeg|jpg|gif|png|ico)'
```

Comment:
@maxrothman mentioned that "there are other URLs where static assets can be hosted, e.g. course assets, uploads, profile images, etc." What are some examples of those URLs? Is this regex going to catch them all?

That aside, it's not clear to me that this is going to add any value beyond what `http.is_safe_url()` gives us. Is it a problem if a user is redirected to an image or other static asset hosted on our site? If it is, then this regex seems pretty weak to me. There are many other static file formats not represented here. If we need to go beyond what `http.is_safe_url()` gets us by inspecting extensions, consider identifying the extensions we want to allow here, instead of trying to individually blacklist them all.